### PR TITLE
Fix querying by unit keys to query by the specific keys instead of the c...

### PR DESCRIPTION
**Pull request by barnabycourt**
...ombinatorial of their values

### CODE REVIEW

**Comment by barnabycourt** [comment]
ok test

File: server/pulp/server/managers/content/query.py [hunk 1]
```diff
@@ -165,23 +166,25 @@ def get_multiple_units_by_keys_dicts(self, content_type, unit_keys_dicts, model_
...
                              None means all fields
-        @type model_fields: None or list of str's
-        @return: tuple of content units found in the content type collection
+        :type model_fields: None or list of str's
+        :return: tuple of content units found in the content type collection
                  that match the given unit keys dictionaries
-        @rtype: (possibly empty) tuple of dict's
-        @raise ValueError if any of the keys dictionaries are invalid
+        :rtype: (possibly empty) tuple of dict's
+        :raise ValueError if any of the keys dictionaries are invalid
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/query.py]: should be ``:raises ValueError:``

http://sphinx-doc.org/domains.html#info-field-lists

File: server/pulp/server/managers/content/query.py [hunk 2]
```diff
@@ -314,29 +317,16 @@ def _build_multi_keys_spec(content_type, unit_keys_dicts):
...
-    @rtype: dict
-    @raise: ValueError if any of the key dictionaries do not match the unique
+    :param content_type: unique id of the content type collection
+    :type content_type: str
+    :param unit_keys_dicts: list of key dictionaries whose key, value pairs can be
+                            used as unique identifiers for a single content unit
+    :type unit_keys_dicts: list of dict
+    :return: mongo db spec document for locating documents in a collection
+    :rtype: dict
+    :raise: ValueError if any of the key dictionaries do not match the unique
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/query.py]: move colon after exception: http://sphinx-doc.org/domains.html#info-field-lists

File: server/pulp/server/managers/content/query.py [hunk 3]
```diff
@@ -346,7 +336,6 @@ def _build_multi_keys_spec(content_type, unit_keys_dicts):
     keys_errors = []
     # spec document valid keys and valid values, used as template to generate
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/query.py]: should this comment also be removed?

File: server/pulp/server/managers/content/query.py [hunk 4]
```diff
@@ -356,13 +345,8 @@ def _build_multi_keys_spec(content_type, unit_keys_dicts):
...
-        if extra_keys or missing_keys:
-            continue
-        # validation passed, store the keys and values in the template
-        for k, v in keys_dict.items():
-            spec_template[k].add(v)
     if keys_errors:
         value_error_msg = '\n'.join(keys_errors)
         raise ValueError(value_error_msg)
-    spec = dict([(k, {'$in': list(v)}) for k, v in spec_template.items()])
+    spec = {'$or': unit_keys_dicts}
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/query.py]: I like that in addition the being more correct, this is a lot easier to understand.

File: server/pulp/server/managers/content/query.py [hunk 5]
```diff
@@ -113,28 +113,29 @@ def get_content_unit_by_keys_dict(self, content_type, unit_keys_dict, model_fiel
...
-        @type model_fields: None or list of str's
-        @return: content unit from the content type collection that matches the
+        :type model_fields: None or list of str's
+        :return: content unit from the content type collection that matches the
                  keys dict
-        @rtype: dict
-        @raise: ValueError if the unit_keys_dict is invalid
-        @raise: L{MissingResource} if no content unit in the content type
+        :rtype: dict
+        :raise: ValueError if the unit_keys_dict is invalid
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/query.py]: should be :raises ValueError:

http://sphinx-doc.org/domains.html#info-field-lists

File: server/pulp/server/managers/content/query.py [hunk 6]
```diff
@@ -113,28 +113,29 @@ def get_content_unit_by_keys_dict(self, content_type, unit_keys_dict, model_fiel
...
+        :raise: L{MissingResource} if no content unit in the content type
```
Comment (mhrivnak) [inline server/pulp/server/managers/content/query.py]: similar

**Pull request merged**